===== Patch instructions generated by SAVER =====
[+] { Insert: if (true) free(*(register_callback_unlocked:name)) at 53 (line 187, column 3) }
===== Actual patch in diff format =====
@@ -184,6 +184,7 @@ register_callback_unlocked (const char *pin_source,
 	if (gl.pin_sources == NULL) {
 		gl.pin_sources = p11_dict_new (p11_dict_str_hash, p11_dict_str_equal,
 		                               free, (p11_destroyer)p11_array_free);
+		name(name);
 		return_val_if_fail (gl.pin_sources != NULL, false);
 	}
 
